Examples/restcomm-olympus/app/src/androidTest/java/org/restcomm/android/olympus/IntegrationTests.java

```diff
@@ -1029,6 +1046,511 @@ public void run() {
+                params.put(RCDevice.ParameterKeys.SIGNALING_DOMAIN, SERVER_HOST + ":" + SERVER_PORT);
+                params.put(RCDevice.ParameterKeys.SIGNALING_USERNAME, CLIENT_NAME);
+                params.put(RCDevice.ParameterKeys.SIGNALING_PASSWORD, CLIENT_PASSWORD);
+                params.put(RCDevice.ParameterKeys.MEDIA_ICE_URL, ICE_URL);
+                params.put(RCDevice.ParameterKeys.MEDIA_ICE_USERNAME, ICE_USERNAME);
+                params.put(RCDevice.ParameterKeys.MEDIA_ICE_PASSWORD, ICE_PASSWORD);
+                params.put(RCDevice.ParameterKeys.MEDIA_ICE_DOMAIN, ICE_DOMAIN);
+                params.put(RCDevice.ParameterKeys.MEDIA_TURN_ENABLED, true);
+                params.put(RCDevice.ParameterKeys.SIGNALING_SECURE_ENABLED, false);
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ENABLE_PUSH_FOR_ACCOUNT, true);
+                params.put(RCDevice.ParameterKeys.DEBUG_JAIN_DISABLE_CERTIFICATE_VERIFICATION, true);
+
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_APPLICATION_NAME, PUSH_APPLICATION_NAME);
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ACCOUNT_EMAIL, PUSH_ACCOUNT);
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_ACCOUNT_PASSWORD, PUSH_PASSWORD);
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_PUSH_DOMAIN, "invalid domain");
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_HTTP_DOMAIN, HTTP_DOMAIN);
+                params.put(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_FCM_SERVER_KEY, "server key");
+
+
+                device.setLogLevel(Log.VERBOSE);
+
+                try {
+                    device.initialize(InstrumentationRegistry.getTargetContext(), params, IntegrationTests.this);
+                } catch (RCException e) {
+                    Log.e(TAG, "RCDevice Initialization Error: " + e.errorText);
+                }
+            }
+        });
+
+        await().atMost(PUSH_TIMEOUT, TimeUnit.SECONDS).until(fieldIn(this).ofType(boolean.class).andWithName("onWarningReceived"), equalTo(true));
+        assertThat(context.get("status-code")).isEqualTo(RCClient.ErrorCodes.ERROR_DEVICE_PUSH_NOTIFICATION_APPLICATION_MISSING.ordinal());
+
+        clientHandler.post(new Runnable() {
+            @Override
+            public void run() {
+                resetPushConfiguration();
+                device.release();
+            }
+        });
+
+        await().atMost(SIGNALING_TIMEOUT, TimeUnit.SECONDS).until(fieldIn(this).ofType(boolean.class).andWithName("deviceReleased"), equalTo(true));
+        assertThat(((RCDevice) context.get("device")).getState()).isEqualTo(RCDevice.DeviceState.OFFLINE);
+        assertThat(context.get("status-code")).isEqualTo(0);
+
+        InstrumentationRegistry.getTargetContext().unbindService(this);
+    }
+
+    @Test(timeout = TC_TIMEOUT)
+    public void deviceInitializeWithPush_DisableEnablePush() throws InterruptedException {
```

Comment:
This seems like a 'valid' scenario, right? If so can we make it 'deviceInitializeWithPushDisableEnablePush_Valid'?

Reply:
yes, done.